### ISSUE TITLE
Exclusion rule using annotatedBy does not work if multiple annotations are present

### ISSUE THREAD

**Issue by NasiaKoutsopoulou** [Kotlin/kotlinx-kover#528]
**Describe the bug**
If we set an exclusion rule using annotatedBy, for annotation that uses AnnotationRetention.BINARY or AnnotationRetention.RUNTIME, but the given function has multiple annotations set, and one of those is AnnotationRetention.SOURCE, exclusion is not working.

**Expected behavior**
We would expect to be able to exclude a function based on a specific annotation, regardless of other annotations being present.

**Reproducer**
```kotlin
@OptIn(ExperimentalMaterialApi::class)
@Composable
fun TestComposable() {
  Text("This is a test composable")
}
```
Note that OptIn is using AnnotationRetention.SOURCE whereas Composable annotation is using AnnotationRetention.BINARY.

Our Kover.gradle contains the following rule:
```kotlin
koverReport {
  filters {
    excludes {
      classes(
         ...
      )
      annotatedBy(
        "androidx.compose.runtime.Composable",
        ...
      )
    }
    ...
  }
```

**Environment**
 - Kover Gradle Plugin version: 0.7.3
 - Gradle version: 8.2.1
 - Kotlin project type: Kotlin/Android
 - Coverage Toolset (if customized in build script): Kover
 
 
 If it feasible to support such cases or is there any workaround for this?

**Comment by shanshin** [comment]
Hi,
in your case, if you remove the `@OptIn` annotation, will function `TestComposable` be excluded from the HTML report?

If so, could you please create a complete producer project, because I can't reproduce the problem locally (the `TestComposable` function successfully excluded from the report)

**Comment by NasiaKoutsopoulou** [comment]
Hey @shanshin.
Apologies for the misunderstanding.

Upon further investigation, it appears that the issue is not related to the additional annotation after all. The problem seems to stem from our composable, the lambda parameter and the associated underlying Java bytecode.

I am attaching some screenshots for additional clarification. Below is our composable:

```kotlin
@Composable
@OptIn(ExperimentalMaterialApi::class)
fun DismissibleUpcomingRequest(
  state: UpcomingTimeOffRequestUIState,
  onActionA: (UpcomingTimeOffRequestUIState, Dp) -> Unit,
  onActionB: () -> Unit
) {
...
}
```

I looked into the HTML report because we were using Danger and the PR reporting seemed off. Turns out, the exclusion of annotated composables is working, but the generated bytecode, related to our lambda parameter, isn't giving us the exclusion we wanted.

Not sure if it's our Compose usage or something up with Kover. 

I am closing this as a false issue report, intending to delve further into identifying the root cause. Should we require support from Kover, I will revisit the matter accordingly.

<img width="300" alt="Screenshot 2024-01-25 at 1 59 53 PM" src="https://github.com/Kotlin/kotlinx-kover/assets/54838695/d55fed68-3324-475e-b70c-7d301fbf34a6">
<img width="229" alt="Screenshot 2024-01-25 at 1 59 45 PM" src="https://github.com/Kotlin/kotlinx-kover/assets/54838695/2798b1fd-0c56-4fd3-ab2f-b9ea854bf345">
